openhands/sdk/agent/agent.py

```diff
@@ -114,8 +52,6 @@ def _configure_bash_tools_env_provider(self, state: ConversationState) -> None:
         Configure bash tool with reference to secrets manager.
         Updated secrets automatically propagate.
         """
-        if not isinstance(self.tools, dict):
```

Comment:
We probably need some defensive code here to check that the `tool_map` as initialized properly? 

We need `agent.initialize()` to be called before `agent.init_state()` for this to work properly. Maybe we should consider calling `agent.initialize()` inside `agent.init_state()` so everything is instantiated and wired with one call

Reply:
done in https://github.com/All-Hands-AI/agent-sdk/pull/356/commits/048adf4c5e64d54072701c48aa1ba0645d4c6c92